config/routes.rb

```diff
@@ -117,11 +117,13 @@
 
     resources :pages do
       get 'share-rows', on: :member, action: 'share_rows'
+      get 'actions', on: :member, action: 'actions'
       get 'featured', on: :collection
 
       resource  :analytics
       resources :actions, only: [:create] do
         post 'validate', on: :collection, action: 'validate'
+        put 'publish', on: :member, action: 'publish'
```

Comment:
I thought this would make it a bit more explicit, but if you think it's better to make it more RESTful then I don't object.